paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -119,6 +119,25 @@ class MKLDNNHandler {
     return mem_p;
   }
 
+  std::shared_ptr<mkldnn::memory> AcquireMemory(
+      const std::vector<int>& dims, const mkldnn::memory::data_type dtype,
+      const mkldnn::memory::format& fmt, void* ptr, const std::string& suffix) {
+    /*Generate key*/
+    auto local_key = key_ + suffix;
+    auto mem_p =
+        std::static_pointer_cast<mkldnn::memory>(dev_ctx_.GetBlob(local_key));
+    if (mem_p == nullptr) {
+      auto md = mkldnn::memory::desc(dims, dtype, fmt);
+
+      mem_p = std::make_shared<mkldnn::memory>(
+          mkldnn::memory::primitive_desc{md, engine_}, ptr);
+      dev_ctx_.SetBlob(local_key, mem_p);
+    } else {
+      mem_p->set_data_handle(ptr);
+    }
+    return mem_p;
+  }
+
```

Comment:
As said above, I think we shouldn't create memory descriptor from parameters like that: `auto md = mkldnn::memory::desc(dims, dtype, fmt);` but use `fwd_pd_->src_desc()` or others instead. This will be possible once we create (or acquire from GetBlob) primitive descriptor in constructor or just before creating memory for primitives.

Reply:
To certain degree this can be done . I will modify this PR to replace AcquireDstMemory with AcquireDstMemoryFromPrimitive like here: https://github.com/PaddlePaddle/Paddle/blob/85914f7a88f748a3883bfd09df810e910cce3089/paddle/fluid/operators/mkldnn/activation_mkldnn_op.cc#L120-L121

Reply:
On second thought. Some more work is to be done here, so I created separated issue for that #19566